src/index.js

```
@@ -1,9 +1,89 @@
+    const describeResponse = await cloudWatchLogsClient.send(
+      describeLogGroupsCommand,
+    );
+    const logGroupExists = describeResponse.logGroups.some(
+      (group) => group.logGroupName === logGroupName,
+    );
+
+    if (!logGroupExists) {
+      const createLogGroupCommand = new CreateLogGroupCommand({ logGroupName });
+      await cloudWatchLogsClient.send(createLogGroupCommand);
+      console.log(`CloudWatch Log Group created: ${logGroupName}`);
+    } else {
+      console.log(`CloudWatch Log Group already exists: ${logGroupName}`);
+    }
+  } catch (error) {
+    if (error.name !== "ResourceAlreadyExistsException") {
+      console.error(`Error ensuring log group exists: ${error.message}`);
+      throw new Error(`Error ensuring log group exists: ${error.message}`);
+    }
+  }
+}
+
+/**
+ * Ensures that the specified CloudWatch log stream exists within a log group, and creates it if it does not.
+ *
+ * @param {string} logGroupName - The name of the log group.
+ * @param {string} logStreamName - The name of the log stream.
+ */
+async function ensureLogStreamExists(logGroupName, logStreamName) {
+  try {
+    const describeLogStreamsCommand = new DescribeLogStreamsCommand({
+      logGroupName,
+      logStreamNamePrefix: logStreamName,
+    });
+    const describeResponse = await cloudWatchLogsClient.send(
+      describeLogStreamsCommand,
+    );
+    const logStreamExists = describeResponse.logStreams.some(
+      (stream) => stream.logStreamName === logStreamName,
+    );
+
+    if (!logStreamExists) {
+      const createLogStreamCommand = new CreateLogStreamCommand({
+        logGroupName,
+        logStreamName,
+      });
+      await cloudWatchLogsClient.send(createLogStreamCommand);
+      console.log(`CloudWatch Log Stream created: ${logStreamName}`);
+    } else {
+      console.log(`CloudWatch Log Stream already exists: ${logStreamName}`);
+    }
+  } catch (error) {
+    if (error.name !== "ResourceAlreadyExistsException") {
```

Fails silently if it is ResourceAlreadyExistsException